app/assets/javascripts/protocols/index.js

```diff
@@ -187,6 +188,34 @@ function initKeywordFiltering() {
   });
 }
 
+function initProtocolPreviewModal() {
+  // Only do this if the repository is public/private
+  if (repositoryType !== "archive") {
+    protocolsTableEl.on("click", "a[data-action='protocol-preview']", function(e) {
+      var link = $(this);
+      $.ajax({
+        url: link.attr("data-url"),
+        type: "GET",
+        dataType: "json",
+        success: function (data) {
+          var modal = $("#protocol-preview-modal");
+          var modalTitle = modal.find(".modal-title");
+          var modalBody = modal.find(".modal-body");
+          modalTitle.html(data.title);
+          modalBody.html(data.html);
+          modal.modal("show");
+          initHandsOnTable($(document));
```

Comment:
Can't you just use `initHandsOnTable(modalBody);` here? We just want to initialize tables inside modal body after all...